Connections.md

```diff
@@ -0,0 +1,10 @@
+# Improved handling of failing connection attempts
+
+In versions prior to .NET 6, in case there is no connection immediately available in the connection pool, a new HTTP request will always spin up a new connection attempt and wait for it. The downside to this is, if it takes a while to establish that connection and another connection becomes available in the meantime, that request would continue waiting for the connection it spawned, hurting latency. In .NET 6.0 [we changed this](https://devblogs.microsoft.com/dotnet/dotnet-6-networking-improvements/#other-http-changes) to process requests on whichever connection that becomes available first, whether that’s a newly established one or one that became ready to handle the request in the meantime.
+
+Unfortunately, the .NET 6.0 implementation turned out to be problematic for some users: a failing connection attempt also fails the request on the top of the request queue, which  may lead to unexpected request failures in [certain scenarios](https://github.com/dotnet/runtime/issues/60654#issue-1030717005). Moreover, if there is a pending connection in the pool that will never become usable eg. because of a misbehaving server or a network issue, new incoming requests associated with it will also stall and may fail unexpectedly.
```

Comment:
I feel like we could elaborate on the second problem a little bit, mention that request cancellation didn't affect connection's creation and one stalled connection could not only "fail new requests unexpectedly" but even lead to massive request timeouts, in case request timeouts were significantly smaller than connect timeouts (especially on Linux)

Reply:
> request cancellation didn't affect connection's creation 

I don't think this point is *that* relevant from user perspective.